src/java.base/share/classes/sun/security/provider/CtrDrbg.java

```diff
@@ -495,13 +495,16 @@ public synchronized void generateAlgorithm(
             // Step 4.1. Increment
             addOne(v, ctrLen);
             try {
-                // Step 4.2. Encrypt
                 cipher.init(Cipher.ENCRYPT_MODE, new SecretKeySpec(k, keyAlg));
-                byte[] out = cipher.doFinal(v);
-
+                // Step 4.2. Encrypt
                 // Step 4.3 and 5. Cat bytes and leftmost
-                System.arraycopy(out, 0, result, pos,
-                        (len > blockLen) ? blockLen : len);
+                if (len > blockLen) {
+                    cipher.doFinal(v, 0, blockLen, result, pos);
+                } else {
+                    byte[] out = cipher.doFinal(v);
+                    System.arraycopy(out, 0, result, pos, len);
+                    Arrays.fill(out, (byte)0);
+                }
```

Comment:
No need to handle encryption output?

Reply:
`out` contains same bytes as the output, and the output is often directly used as key materials.